### PR TITLE
Model domain

### PR DESCRIPTION
**Pull request by cezarypiatek**
I've added complete logic for poker hands coparing

### CODE REVIEW

File: src/Blef.GameLogic/PokerHands/!PokerHandHierarchy.txt [hunk 1]
```diff
@@ -1,11 +1,11 @@
...
-Flush - kolor (kolor)
-FullHouse - full (wartość, wartość)
-ThreeOfKind - trójka (wartość)
-HighStraight - duży strit
-LowStraight - mały strit
-TwoPairs - dwie pary (wartość, wartość)
-Pair - para (wartość)
-HighCard - wysoka karta (wartość)
\ No newline at end of file
+﻿HighStraightFlush - duży poker (kolor) 1 000 000 000 *S
```
Comment (ArturWincenciak) [inline src/Blef.GameLogic/PokerHands/!PokerHandHierarchy.txt]: What does means 

> 1 000 000 000 *S

?

File: src/Blef.GameLogic/PokerHands/!PokerHandHierarchy.txt [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
+LowStraightFlush -  mały poker (kolor) 100 000 000 *S
+FourOfKind - kareta (wartość) 10 000 000 *R
+Flush - kolor (kolor) 1 000 000 * S
+FullHouse - full (wartość, wartość) 100 000 + 10R1 +R2
```
Comment (ArturWincenciak) [inline src/Blef.GameLogic/PokerHands/!PokerHandHierarchy.txt:5]: What does means 

> 100 000 + 10R1 +R2

?

File: src/Blef.GameLogic/PokerHands/Flush.cs [hunk 3]
```diff
@@ -17,9 +17,6 @@ public override bool IsOnTable(Table table)
             return table.HasSuit(suit);
         }
 
-        public override bool IsStrongerThan(PokerHand otherPokerHand)
-        {
-            throw new System.NotImplementedException();
-        }
+        protected override int Value => 1000000 * GetSuitValue(suit);
```
Comment (ArturWincenciak) [inline src/Blef.GameLogic/PokerHands/Flush.cs]: What are these strange mathematics with large numbers?

File: src/Blef.GameLogic/PokerHands/FullHouse.cs [hunk 4]
```diff
@@ -23,9 +23,6 @@ public override bool IsOnTable(Table table)
             return firstRankCount >= 3 && secondRankCount >= 2;
         }
 
-        public override bool IsStrongerThan(PokerHand otherPokerHand)
-        {
-            throw new System.NotImplementedException();
-        }
+        protected override int Value => 100000 + 10 * GetRankValue(first) +  GetRankValue(second);
```
Comment (ArturWincenciak) [inline src/Blef.GameLogic/PokerHands/FullHouse.cs]: What are these strange mathematics with large numbers?

**Review by ArturWincenciak — COMMENTED** [review]
What are these strange mathematics with large numbers?

File: src/Blef.GameLogic/PokerHands/!PokerHandHierarchy.txt (hunk 2 above)
Comment (cezarypiatek) [reply src/Blef.GameLogic/PokerHands/!PokerHandHierarchy.txt:5]: R means Rank value and S means Suit value. I've assigned value to each poker hand to establish hierarchy. Each poker hand  value is const value or function of components (suit or rank and basic value). Basic value help us to tell apart poker hands on different level.
FullHouse consists of two ranks (ex first rank = three kings and second rank = two queens) so the full value of FullHouse is basic value (100 000) + 10* firstRank* rankValue + secondRank*rankValue. Simply speaking R1 is the first rank (which is stronger) and R2 is the second rank. This approach allows us to distinguish poker hands with different level (ex HighCard vs Pair) and poker hands on the same level with different rank/suit as well.

**Comment by Enumix** [comment]
Przydałyby się jakieś testy na sprawdzenie poprawności porównywania - jeżeli dobrze widzę, to nie ma żadnych? Później będzie się można zastanowić nad jakimś ładniejszym sposobem porównywania, obecny sposób jest sprytny, przez co niezbyt czytelny :)

**Comment by kmorcinek** [comment]
Myślę tak samo jak Enumix w kwestii czytelności, tudzież nieczytelności. Kiedyś bym to zmienił na bardziej domenowe.

Btw @Enumix pisać Ci Wiktor czy Enumix?

**Comment by kmorcinek** [comment]
Were conflicts. I merged by hand and was fine. Closing.

**Pull request closed without merging**